### ISSUE TITLE
Restoring sogo backup with different usernames

### ISSUE THREAD

**Issue by jaapvdv** [mailcow/mailcow-dockerized#2101]
I am currently preparing to migrate from my current mailserver (Nethserver 6 in an ESXi VM) to mailcow-dockerized running in a ProxMox VE 5.3 VM.

Migrating email has been (and still is) incredibly easy using mailcow's built in synchronization tasks.

Restoring the SOGo 3 backup from Nethserver 6 however is more challenging. 

Taking my own account as an example: 
- my username in Nethserver 6 is 'jaapvdv';
- in mailcow my username is 'jaap@domain.com'.

Does anyone have a suggestion on how to succesfully restore/migrate this data?

**Comment by andryyy** [comment]
Hi,

Hm, you could rename your user on your Nethserver instance by running `sogo-tool rename-user jaapvdv jaap@domain.com`.

After that, you could backup the users data with `sogo-tool backup /tmp/my_sogo_stuff jaap@domain.com`.

You can then copy the backup to your mailcow and move it to data/conf/sogo, to be able to access it inside the SOGo container.

Enter the container with `docker-compose exec sogo-mailcow /bin/bash`, run `su -s /bin/bash sogo` and run the restore command, which would be something like `/usr/sbin/sogo-tool restore ALL /etc/sogo/my_sogo_stuff jaap@domain.com`.

**Comment by jaapvdv** [comment]
Thanx André, I'll give that a try and report back

**Comment by jaapvdv** [comment]
Hmm, to be safe I added a new user to my my current server with username 'jaapvdv2'. I copied a large calendar and addresbook to the new account, and:

`su - sogo -s '/bin/bash' -c '/usr/sbin/sogo-tool rename-user jaapvdv2 jaap@domain.com'`

seems to work, so does (obviously):

`su - sogo -s '/bin/bash' -c '/usr/sbin/sogo-tool backup /tmp/sogo ALL'`

But if I list the contents of the folder /tmp/sogo there is no export file for 'jaap@domain.com' (there is one for 'jaapvdv2' though).

Checking the file contents clearly show only jaapvdv2 is shown as uid and not jaap@eisergon.nl so rename the file to jaap@domain.com and importing it into mailcow will not be useful.

Conclusion for now: might/should work if the uid is actually changed, not a mailcow issue but a Nethserver (or SOGo 3?) one!

**Comment by jaapvdv** [comment]
Tried something else; renamed the backup for the user to jaap@domein.com

Edited the file;
changed uid and c_uid to jaap@eisergon.nl

Restored the file, complains about:
`ERROR(+[GCSFolderManager` defaultFolderManager]): default 'OCSFolderInfoURL' is not configured.
missing folder '/Users/jaapvdv/Contacts/personal' could not be recreated`

**Comment by andryyy** [comment]
Did you change to sogo? `su -s /bin/bash sogo`?

**Comment by jaapvdv** [comment]
Good one, forgot about that.

Executing 'sogo-tool restare -f ALL' to restore the manually modified backup file no longer results in the earlier errors. The imported data however, is not visible in SOGo.

The earlier error messages indicate missing folder names, is mailcow sogo using these folder names/paths or different ones?
It appears sogo-tool is restoring just fine now, but sogo appears not be reading data from the restored 'folders'

**Comment by andryyy** [comment]
I think this may happen due to renaming the user in the file. Did you miss something perhaps? 
You could try to backup a mailcow user and compare both files (the renamed one and the "correct" backup). Any differences in the format?

**Comment by jaapvdv** [comment]
Hi André,

Yes, I definitely did not change all relevant parts of the backup file.

But with your suggestion I have been able to succesfully restore all calendar and contact  info.

It does however require some manual modifications to the backup files before import.

I compared my NethServer SOGo backup file (NS) to a 'blank' Mailcow SOGo backup file (MC).

First thing is to replace the LDIF record from MC to NS (starting at line 1 containing only **{** until the line  only containing **";**.

Furthermore all occurences of usernames in the NS file have to be replaced with the correct usernames from MC (that is not only the username of the 'current' user but also those of shared calendars and addressbooks).

For Calendars this means changing to /Users/user@domain.tld/Calendar

For Addresbooks this means changing to /Users/user@domain.tld/Contacts
